PluginGenerator/JarManifestReader.cs

```diff
@@ -0,0 +1,86 @@
+﻿/*
+ * SonarQube Roslyn SDK
+ * Copyright (C) 2015-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+
+using System;
+using System.Collections.Generic;
+
+namespace SonarQube.Plugins
+{
+    /// <summary>
+    /// Reads a valid v1.0 Jar Manifest.
+    /// See http://docs.oracle.com/javase/6/docs/technotes/guides/jar/jar.html#JAR%20Manifest
```

Comment:
Simple reader to extract key-value pairs from an existing manifest file.
Required so we can copy single- and multi-line properties from the manifest in the template jar.